### PR TITLE
[WIP] Devguide A1->A2 upgrade chapter (candidate)

### PR DESCRIPTION
**Pull request by wardbell**

### CODE REVIEW

**Comment by teropa** [comment]
@wardbell @naomiblack This PR contains the PhoneCat upgrade tutorial, which is the part that's furthest along. It also has the outline and some preliminary content for the wider Upgrade section. My question is do you want me to yank out everything except the tutorial from this PR?

**Comment by wardbell** [comment]
I haven't looked yet. I want to. Hope Naomi beats me to it

**Comment by naomiblack** [comment]
hosted for review at https://ngdocsdev.firebaseapp.com/docs/ts/latest/guide/upgrade.html

still troubleshooting some issues with doc shredder but text is ready to review...

**Comment by naomiblack** [comment]
I think you missed a couple of the examples when you pulled phonecat out? I get errors from L164 of `upgrade.jade` and don't have the source under `examples` for 
```
+makeJson('upgrade/ts/typescript-conversion/tsconfig.json', null, 'tsconfig.json')
```
or on L174 for 
```
+makeJson('upgrade/ts/typescript-conversion/package.json', {paths: 'scripts'}, 'package.json', {otl: /(\"tsc.*)/})
```

File: public/docs/ts/latest/guide/upgrade.jade [hunk 1]
```diff
@@ -0,0 +1,1291 @@
...
+  
++makeExample('upgrade/ts/typescript-conversion/test/test_helper.ts', null, 'test/test_helper.ts')
+
+:marked
+  For Karma's SystemJS support we'll use a shim file that will tweak the way
+  files get loaded, so that it happens through SystemJS:
+  
++makeExample('upgrade/ts/typescript-conversion/test/karma_test_shim.js', null, 'test/karma_test_shim.js')
+
+.alert.is-important The shim is likely to be replaced by improved tooling, but is needed in the current, early versions of Angular 2.
```
Comment (bradlygreen) [inline public/docs/ts/latest/guide/upgrade.jade]: Can you delete "early versions" from this sentence?

File: public/docs/ts/latest/guide/upgrade.jade [hunk 2]
```diff
@@ -0,0 +1,1291 @@
...
+  * We've removed the `vm.` prefix from all expressions.
+  * Just like we did in the phone list, we've replaced `ng-src` with property
+    bindings for the standard `src`.
+  * We're using the property binding syntax around `ng-class`. Though Angular 2
+    does have a very similar `ngClass` as Angular 1 does, its value is not
+    magically evaluated as an expression. In Angular 2 we always specify
+    in the template when an attribute's value is a property expression, as opposed
+    to a literal string.
+  * We've replaced `ng-repeat`s with `*ngFor`s.
+  * We've replace `ng-click` with an event binding for the standard `click`.
```
Comment (bradlygreen) [inline public/docs/ts/latest/guide/upgrade.jade]: Change "replace" to "replaced"

**Comment by teropa** [comment]
Thank you @naomiblack and @bradlygreen! Just pushed an update addressing your comments.

**Comment by naomiblack** [comment]
This is live on the site and now in master. Thank you!

**Pull request closed without merging**